nntrainer/opencl/opencl_program.cpp

```diff
@@ -95,29 +99,36 @@ bool Program::GetProgramInfo(cl_device_id device_id) {
   }
 
   // getting the kernel names
-  char kernel_names[kernel_names_size];
-  error_code = clGetProgramInfo(program_, CL_PROGRAM_KERNEL_NAMES,
-                                kernel_names_size, kernel_names, nullptr);
+  // char kernel_names[kernel_names_size];
+  std::vector<char> kernel_names(kernel_names_size);
+
+  error_code =
+    clGetProgramInfo(program_, CL_PROGRAM_KERNEL_NAMES, kernel_names_size,
+                     kernel_names.data(), nullptr);
 
   if (error_code != CL_SUCCESS) {
     ml_loge("Failed to get program kernel names. OpenCL error code: %d. %s",
             error_code,
             (GetProgramBuildInfo(device_id, CL_PROGRAM_BUILD_LOG)).c_str());
     return false;
   } else {
-    ml_logi("Saving kernel binary for: %s", std::string(kernel_names).c_str());
+    ml_logi("Saving kernel binary for: %s",
+            std::string(kernel_names.data()).c_str());
   }
 
   // Read the binary
   size_t binaries_ptr_alloc_size = sizeof(unsigned char *) * num_devices;
-  unsigned char *binaries_ptr[num_devices];
+
+  // unsigned char *binaries_ptr[num_devices];
+  std::vector<unsigned char *> binaries_ptr;
 
   for (unsigned int i = 0; i < num_devices; ++i) {
-    binaries_ptr[i] = new unsigned char[binaries_size[i]];
+    binaries_ptr.push_back(new unsigned char[binaries_size[i]]);
```

Comment:
I couldn't find a memory release code for this variable. Is it deleted somewhere else later on?

Reply:
it is released in the latter part!

https://github.com/nnstreamer/nntrainer/blob/656a231787c426e021f0c58479d2ffaabf3558b2/nntrainer/opencl/opencl_program.cpp#L155-L158